zilliqa/src/crypto.rs

```diff
@@ -20,20 +22,44 @@ use sha3::{Digest, Keccak256};
 
 /// The signature type used internally in consensus, to e.g. sign block proposals.
 #[derive(Debug, Clone, Copy, PartialEq)]
-pub struct NodeSignature(bls_signatures::Signature);
+pub struct NodeSignature(Signature<Bls12381G2Impl>);
 
 impl NodeSignature {
     pub fn identity() -> NodeSignature {
-        NodeSignature(G2Affine::identity().into())
+        // Default to Basic signatures - it's the normal signature.
+        NodeSignature(Signature::<Bls12381G2Impl>::Basic(
+            blsful::inner_types::G2Projective::identity(),
+        ))
     }
 
     pub fn from_bytes(bytes: &[u8]) -> Result<NodeSignature> {
-        Ok(NodeSignature(bls_signatures::Signature::from_bytes(bytes)?))
+        // Default to Basic signatures - it's the normal signature.
+        Ok(NodeSignature(Signature::<Bls12381G2Impl>::Basic(
+            blsful::inner_types::G2Projective::from_compressed(bytes.try_into()?)
```

Comment:
Just to double check - You verified that `bls_signatures::Signature::from_bytes` was treating the bytes as a compressed signature?

Reply:
Yes, there's a check in `bls_signatures::Signature::from_bytes`:
```
    if raw.len() != G2_COMPRESSED_SIZE {
        return Err(Error::SizeMismatch);
    }
```

Reply:
But I don't seem to find a length check in the underlying library, so I've added length checks in ours.